packages/core/src/types.ts

```diff
@@ -705,6 +713,8 @@ export interface HistoryStateNode<TContext> extends StateNode<TContext> {
   target: StateValue | undefined;
 }
 
+export type SchemaValidator<T> = (value: any) => value is T;
```

Comment:
is this used anywhere?

Reply:
It isn't; I used it when experimenting with the other (validation function) approach 😅 

We can probably make use of this in future work if we do some of the "design by contract" stuff, but I'll delete it